firecrown/cosmosis/run.py

```diff
@@ -0,0 +1,207 @@
+import pathlib
+import os
+import sys
+
+# Locate the path to this directory
+# For fiddly reasons that make a great deal of sense in an entirely
+# different context to this one, we pass cosmosis a full path
+# to the interface file
+THIS_DIRECTORY = pathlib.Path(__file__).parent.resolve()
+COSMOSIS_INTERFACE = str(THIS_DIRECTORY.joinpath('interface.py'))
+
+
+def run_cosmosis(config, data):
+    """Run CosmoSIS on the problem.
+
+    This requires the following parameters 'sampler' section
+    of the config:
+
+      sampler - name of sampler to use, e.g. emcee, multinest, grid, ...
+      output - name of file to save to
+
+
+    Parameters
+    ----------
+    config : dict
+        Configuration info, usually read directly from the YAML file
+    data : dict
+        The result of calling `firecrown.config.parse` on an input YAML
+        config.
+
+    """
+    from cosmosis.main import run_cosmosis
+
+    # Extract the bits of the config file that
+    # cosmosis wants
+    ini = _make_cosmosis_config(config['sampler'])
+    values = _make_cosmosis_values(config['parameters'])
+    pool = _make_parallel_pool(config['sampler'])
+    pipeline = _make_cosmosis_pipeline(data, values, pool)
+
+    # Actually run the thing
+    run_cosmosis(None, pool=pool, ini=ini, pipeline=pipeline, values=values)
+
+    if pool is not None:
+        pool.close()
+
+
+def _make_parallel_pool(config):
+    """ Set up a parallel process pool.
+
+    Parameters
+    ----------
+
+    config: dict
+        Sampler configuration section of the input
+
+    Will look for the 'mpi' command in the parallel section
+
+    Returns
+    -------
+
+    pool: CosmoSIS MPIPool object
+        parallel process pool
+    """
+
+    from cosmosis.runtime.mpi_pool import MPIPool
+
+    # There is a reason to make the user actively
+    # request to use MPI rather than just checking -
+    # on many systems, including, importantly, NERSC,
+    # trying to import MPI when not running under the
+    # MPI environment will cause a crash
+    use_mpi = config.get('mpi', False)
```

Comment:
The doc string said it looks for 'mpi' in the parallel section, but it is pulling it out of the main thing here. Is there a reason or did I miss something?

Reply:
fixed this